apis/core/bootstrap/shapes/dimension.ts

```diff
@@ -1,6 +1,44 @@
 import { shape } from '@cube-creator/core/namespace'
-import { turtle } from '@tpluscode/rdf-string'
+import { sparql, turtle } from '@tpluscode/rdf-string'
+import { lindasQuery } from '../lib/query'
 import { dash, hydra, rdf, rdfs, schema, sh, qudt } from '@tpluscode/rdf-ns-builders'
+import namespace from '@rdfjs/namespace'
+
+const sou = namespace('http://qudt.org/vocab/sou/')
+
+const unitsQuery = sparql`
+CONSTRUCT 
+  { 
+    ?c a ${hydra.Collection} .
+    ?c ${hydra.member} ?unit .
+    ?unit ${rdfs.label} ?name .
+  }
+WHERE
+  { BIND(iri("http://app.cube-creator.lndo.site/units") AS ?c)
+    { SELECT  ?unit ?name
+      WHERE
+        { GRAPH <https://lindas.admin.ch/ontologies>
+            { ?unit  a ${qudt.Unit} ;
+                     ${rdfs.label} ?label ;
+                     ${qudt.ucumCode} ?ucumCode.
+              OPTIONAL
+                { ?unit  ${qudt.symbol}  ?symbol }
+
+              BIND(concat(?label, " (", coalesce(str(?symbol), str(?ucumCode), "?"), ")") AS ?name)
+              BIND(strlen(str(?ucumCode)) AS ?ucumCount)
+              
+              FILTER ( lang(?label) = "en" )
+              FILTER NOT EXISTS {?unit ${qudt.unitOfSystem} ${sou.SOU_IMPERIAL} . }
+              FILTER NOT EXISTS {?unit ${qudt.unitOfSystem} ${sou.SOU_USCS} . }
+              FILTER NOT EXISTS {?unit ${qudt.unitOfSystem} ${sou.SOU_CGS} . }
+              FILTER NOT EXISTS {?unit ${qudt.unitOfSystem} ${sou.SOU_PLANCK} . }
+              FILTER NOT EXISTS {?unit a ${qudt.CurrencyUnit} . }
+            }
+        }
+      ORDER BY ASC(?ucumCount) ASC(?name)
+    }
+  }
```

Comment:
This query still returns over 1300 results, much of which are derived units?

I don't see a good to way to filter more. Should we dynamically query by the typed input?

Reply:
It works reasonably fast as is. But yes, this might be useful, please propose in an  issue how to optimize this.